src/gui/attributetable/qgsdualview.cpp

```diff
@@ -835,6 +834,15 @@ void QgsDualView::hideEvent( QHideEvent *event )
 {
   Q_UNUSED( event )
   saveRecentDisplayExpressions();
+
+  // Better to save settings here than in destructor. This this last can be called after a new
+  // project is loaded. So, when Qgis::ProjectFlag::RememberAttributeTableWindowsBetweenSessions is set,
+  // a new QgsDualView is created at project loading and we restore the old settings before saving the
+  // new one.
+  // And also, we override close event to just hide in QgsDockableWidgetHelper::eventFilter, that's why hideEvent
+  QgsSettings settings;
+  settings.setValue( QStringLiteral( "/qgis/attributeTable/splitterState" ), mConditionalSplitter->saveState() );
+  settings.setValue( QStringLiteral( "/qgis/attributeEditor/splitterState" ), mAttributeEditorViewSplitter->saveState() );
```

Comment:
could you create the new settings with the API? Since there is no ByteArray settings, you can use `QgsSettingsEntryVariant`.

Reply:
Done.

2 questions:
- I made them private because there is no need to be public. Is that OK?
- The 2 settings are now displayed in settings advanced panel but they are not mean to be edited by user. Is there a way of making them "Hidden" ?

Reply:
gentle ping. Is that OK for you?

Reply:
Good question regarding accessibility. I've always put everything public without python exposure though. I'm a bit wondering how it works in the editor. But if it works, that's fine.

Regarding hiding the setting, there is no such thing for now. What about grouping window states in a node group?

Reply:
> Good question regarding accessibility. I've always put everything public without python exposure though. I'm a bit wondering how it works in the editor. But if it works, that's fine.

It works

> What about grouping window states in a node group?

Why not, I'm gonna add it

Reply:
Hummmm... There is also an attribute-table [group](https://github.com/qgis/QGIS/blob/9921db1e6ff9ad0a7481c66a434c5445fabce1df/src/core/settings/qgssettingstree.h#L67) which could have one of the property.

And I would have expected the attribute-table group to be under gui.

What do you think, we keep a window-state group? under gui? with the 2 properties?

Reply:
I'd go with a new group under gui.
We can move the other ones later...

Reply:
Done 64cf3da49ad